storage-proofs/src/hasher/pedersen.rs

```diff
@@ -220,14 +273,18 @@ impl HashFunction<PedersenDomain> for PedersenFunction {
 }
 
 impl LightAlgorithm<PedersenDomain> for PedersenFunction {
+    // Returns the internally stored digest's x-coordinate. If you are using `PedersenFunction` as
+    // an updatable hasher, you should call `self.finalize()` instead of `self.hash()` to retrieve
+    // the hash digest.
     #[inline]
     fn hash(&mut self) -> PedersenDomain {
-        self.0.into()
+        self.curr_hash.into_xy().0.into()
     }
 
+    // Resets any internal state associated with this hasher.
     #[inline]
     fn reset(&mut self) {
-        self.0 = Fr::from_repr(FrRepr::from(0)).expect("failed 0");
+        *self = PedersenFunction::default();
```

Comment:
lets do this manually to avoid locking on the settings as reset is called a lot in the merkle treed hahsing

Reply:
Oh right, good point!